### PR TITLE
Improved glob pattern matching.

### PR DESCRIPTION
**Pull request by ncruces**
As discussed in #36.

### CODE REVIEW

**Comment by ncruces** [comment]
I signed it!

File: extglob.go [hunk 1]
```diff
@@ -1,25 +1,179 @@
...
+		case '+':
+			if err := c.compileSubExpression("(?:", ")+", "\\+"); err != nil {
+				return err
+			}
+		case '@':
+			if err := c.compileSubExpression("(?:", ")", "\\@"); err != nil {
+				return err
+			}
+		case '!':
+			if err := c.compileSubExpression("(?~", ")", "\\!"); err != nil {
```
Comment (ncruces) [inline extglob.go]: Negative matches are not actually supported. Go regexp supports neither negative lookahead `(?!)`, nor the Ruby's absence operator `(?~)` that we'd need to implement this. This returns an `ErrInvalidPerlOp`.

File: firebase.go [hunk 2]
```diff
@@ -33,7 +33,7 @@ type FirebaseManifest struct {
 
 func (mf FirebaseManifest) processRedirects(w http.ResponseWriter, r *http.Request) (bool, error) {
 	for _, redirect := range mf.Redirects {
-		pattern, err := CompileExtGlob(redirect.Source)
+		pattern, err := CompileExtGlob("/" + strings.TrimPrefix(redirect.Source, "/"))
```
Comment (ncruces) [inline firebase.go]: To be compatible with Firebase, patterns should start with a `/`, and be matched against `r.URL.Path` (which also starts with a `/`).

**Comment by surma** [comment]
10/10 for the test suite! That gives me a lot of confidence. Well done.

**Pull request merged**